pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -311,8 +282,15 @@ impl InMemoryLayer {
             .build();
 
         let inner = self.inner.read().await;
-        let reader = inner.file.block_cursor();
 
+        // Determine ValueReads
+        let mut reads: HashMap<Key, Vec<ValueRead>> = HashMap::new();
+        struct ValueRead {
+            entry_lsn: Lsn,
+            pos: u32,
+            len: u32,
+            value_buf: Mutex<Result<Vec<u8>, Arc<std::io::Error>>>,
```

Comment:
Because

```rust
 struct ChunkReadDestination<'a> {
    value_read: &'a ValueRead,
    ...
```

can't be a `&mut ValueRead`, that would mean we have multiple mutable aliases.

We don't in practice, because the merged reads are performed sequentially.
I guess I can use `.try_lock().unwrap()` in those places.